### DISCUSSION TITLE
Understanding Negation - Project Ideas

### DISCUSSION THREAD

**Discussion by eduardoblanco** [eduardoblanco/csc696H-Spring24#5] [Ideas]
We will all post here ideas for projects about negation, even if we don't want to pursue them. It should help finding team mates and refining problems.

The sooner you get started the better.

**Comment by mayanks43** [comment]
Paraphrasing to improve negation handling
================================
- We build a dataset of paraphrases of negated sentences. Using this dataset, we can learn a paraphrase generator and utilize the generated paraphrases for only negated sentences as extra input to models that work on many NLU tasks such as sentiment analysis, translation, etc. and check for performance improvement. 
- The idea is basically just replacing affirmative interpretations with paraphrases in the discussed paper "[Leveraging Affirmative Interpretations from Negation Improves Natural Language Understanding](https://aclanthology.org/2022.emnlp-main.393)". 
- One way to generate paraphrases is ChatGPT. There are also many paraphrase datasets already existing, some are [here](https://paperswithcode.com/datasets?task=paraphrase-generation&lang=english). We can filter these datasets for sentences with negation and utilize them.

**Reply by eduardoblanco** [reply]
Just to clarify, the output of backtranslating can be assumed to be a paraphrase, correct? What the paper does is rename "paraphrase without negation" as "affirmative interpretation." So, if you can paraphrase a sentence with negation without using negation OR paraphrase a sentence without negation using negation, you are getting more pairs.

I think at first we looked at paraphrasing datasets, but the paraphrases very rarely introduced or got rid of the negation. That said, any method to get more pairs could potentially be an improvement.

**Reply by mayanks43** [reply]
You are right about this work already being about using paraphrases in some form. The paper limits the paraphrases to affirmative interpretations. I think using the original interpretation itself (not the original sentence) might've achieved the same objective (of improving negation handling in NLU tasks) based on the results.

**Comment by mayanks43** [comment]
Affirmative Interpretations for language X/multiple languages
=============================================
* This is probably a low-hanging fruit (may not be conference-worthy). Expand this work to another language/languages since this work only applies to English.
* We would need negation cues for the target language/languages. We already know two languages - Spanish and Norwegian. Perhaps, if the student knows another language, they could try to create negation cues for that language.
* We already have parallel corpora for most languages and google translate works for many languages.
* We would also need easily-available NLU tasks in target languages to improve upon.

**Reply by eduardoblanco** [reply]
Exploring NLU tasks in other languages would certainly be worthwhile. But we could also use another language as a proxy to get more en-en pairs? The Norwegian and Spanish sentences are currently disregarded after backtranslating.

Now that I am thinking a bit deeper and we have better models, it seems naive to not train with the affirmative interpretations in multiple languages rather than only English. Backtranslating is noisy, but parallel corpora is a true "gold standard" (or at least much more reliable than backtranslating. I see no reason to not consider the affirmative interpretation in any language (despite we never did).

**Reply by mayanks43** [reply]
> I see no reason to not consider the affirmative interpretation in any language (despite we never did).

In that case, whatever final model (such as the sentiment model) we are passing this affirmative interpretation to, would need to be a multilingual model.

**Comment by eduardoblanco** [comment]
Can you make the decoder of a text generator not use negation (== have a hard rule that removes negation from the vocabulary). That may be a good strategy to automatically generate as many affirmative interpretations as you want (without needing parallel corpora, a negation cue detector, or a machine translation system. In other words: Can you get a system to paraphrase to never use negation? If you can, you have a better way to generate affirmative interpretations. Here is [a paper](https://aclanthology.org/2021.acl-long.9/) that makes sure that some tokens are in the input (what you would need is to ban negation cues!). If interested on this, look for work on _contrained decoding_.

**Comment by mayanks43** [comment]
Automatic generation of dataset for measuring robustness of models to edits
=========================================================
Perhaps, there is already a paper doing this somewhere but I couldn't find it. This proposal isn't concerned with negation but it could be modified to be for negation.

The idea is to generate a model that tries to understand a model's consistency to edits. We are looking to generate consistency scores for models for various kinds of edits. Since there is no negation, there is no need to worry about implications of negations in the questions. We will generate a dataset similar to CondaQA but will generate it automatically. Here are a possible steps to do that:
1. Collect paragraphs from Wikipedia, or anywhere else.
2. Given a sentence, use some SOTA question generation algorithm to generate a question for the given sentence. Now, based on the type of question, we can perform many kinds of edits. However, we would need to know the answer to these questions in the paragraph. We'd have to stick with question generation methods that start from the answer.
3. Where edit: Find entities in the sentence. Swap entities with a similar entity of the same type. For example, in the sentence: "He was exiled to the island of Elba, between Corsica and Italy" replace "Elba" with "St. Helena".
4. When edit: Find time entities, identify and swap to some other number/time.
5. What/Who/Which edit: Find other entities, identify and swap to some other entity of that type.
7. How and Why edits are harder as they require more long form answers than just Yes/No. Perhaps, we can find paragraph that answers the question and modify key points. Answering such questions would require sequence-to-sequence models.

We can, then, evaluate this dataset on various SOTA models for "consistency" to edits. The issue is that since this automatically generated data and could be noisy, it may not be possible to use this as gold truth.

EDIT: Found something similar: https://aclanthology.org/P19-1610.pdf.

**Comment by mayanks43** [comment]
Understanding if negation is still an issue in the LLM world
==============
This idea is after discussion with @brianrez. We want to understand if negation is still a problem in LLMs. The [previous understanding work](https://aclanthology.org/2022.acl-short.81/) seems to be only done on Roberta. Another interesting question to answer would be what type of negation cues cause the most confusion for these models. Expanding this analysis to more languages than English should result in a good paper.
